packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -101,6 +100,17 @@ export class AwsCliCompatible {
       }
     }
 
+    if (!region && this.isEc2) {
+      debug('Looking up AWS region in the EC2 Instance Metadata Service (IMDS).');
+      const imdsOptions = {
+        httpOptions: { timeout: 1000, connectTimeout: 1000 }, maxRetries: 2,
+      };
+      const metadataService = new AWS.MetadataService(imdsOptions);
+      const token = await getImdsV2Token(metadataService);
```

Comment:
Please rewrite the helpers below to properly throw an error, and then `try/catch` and log any errors here.

Reply:
I rewrote it as such:

 ```
     const token = await getImdsV2Token(metadataService)
        .catch((error) => {
          debug(`No token returned from IMDS; Error: ${error}`)
          return undefined;
        });
      region = await getRegionFromImds(metadataService, token)
        .catch((error) => {
          debug(`No Instance Identity Document returned from IMDS; Error: ${error}`)
          return undefined;
        });
```